src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -124,19 +149,39 @@ protected override void OnNextCore(Block block)
         /// </para>
         /// TODO: consider moving the relay logic to the <see cref="LoopSteps.ProcessPendingStorageStep"/>.
         /// </remarks>
-        public void RelayWorker()
+        private async Task RelayWorkerAsync()
         {
             this.logger.LogTrace("()");
 
-            this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
+            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
             {
                 this.logger.LogTrace("()");
 
+                try
+                {
+                    // Wait until a new block is added to the queue.
+                    this.blockEnqueued.Wait(this.nodeLifetime.ApplicationStopping);
```

Comment:
Why not use a BlockingCollection?

Reply:
` this.blockEnqueued.Wait()` does not block?

Reply:
It does and I think we want to remove it.

Reply:
Why would we remove it? The whole reason for this is to block the task until we get a signal that new block arrived.

Reply:
because 
- it is not awaitable
- it does not implement the 5 second timer

well we could leave it and add linked cancellation token with 5 second timer and we could run that with Task.Run and not have it awaitable

but it seems inferior to have it fully async and not to occupy a thread

Reply:
@noescape00 when we block we hold a thread from the system, in this case its just 1 thread but generally its bad practice.